cmd/telepilot/main.go

```diff
@@ -2,26 +2,133 @@ package main
 
 import (
 	"context"
+	"fmt"
 	"log" // TODO: Consider using slog.
+	"os"
+	"path"
 
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/credentials/insecure"
+	"github.com/google/uuid"
+	"github.com/urfave/cli/v3"
 
-	pb "go.creack.net/telepilot/api/v1"
+	"go.creack.net/telepilot/pkg/apiclient"
+	"go.creack.net/telepilot/pkg/tlsconfig"
 )
 
+//nolint:funlen // Acceptable for CLI definition.
 func main() {
-	conn, err := grpc.NewClient("localhost:9090", grpc.WithTransportCredentials(insecure.NewCredentials()))
-	if err != nil {
-		log.Fatalf("grpc new client: %s.", err)
-	}
-	defer func() { _ = conn.Close() }() // Best effort.
+	var client *apiclient.Client
+	var jobID uuid.UUID
 
-	client := pb.NewTelePilotServiceClient(conn)
+	jobIDArg := &cli.StringArg{
+		Name:      "<job_id>",
+		UsageText: "<job_id>",
+		Min:       1,
+		Max:       1,
+	}
+	parseJobID := func(_ context.Context, cmd *cli.Command) error {
+		id, err := uuid.Parse(cmd.Args().First())
+		if err != nil {
+			return fmt.Errorf("invalid <job_id>, not a uuid: %w", err)
+		}
+		jobID = id
+		return nil
+	}
 
-	ctx := context.Background()
+	cmd := &cli.Command{
+		// Before any command, load the certs and connect to the server.
+		Before: func(_ context.Context, cmd *cli.Command) error {
+			certDir := cmd.String("certs")
+			user := cmd.String("user")
+			tlsConfig, err := tlsconfig.LoadTLSConfig(
+				path.Join(certDir, "client-"+user+".pem"),
+				path.Join(certDir, "client-"+user+"-key.pem"),
+				path.Join(certDir, "ca.pem"),
+				true,
+			)
+			if err != nil {
+				return fmt.Errorf("load tls config for %q from %q: %w", user, certDir, err)
+			}
+			// TODO: Consider making the addr a flag.
+			c, err := apiclient.NewClient(tlsConfig, "localhost:9090")
+			if err != nil {
+				return fmt.Errorf("connect: %w", err)
+			}
+			client = c
+			return nil
+		},
+		// After any command, disconnect.
+		After: func(_ context.Context, _ *cli.Command) error {
+			if client != nil {
+				return client.Close()
+			}
+			return nil
+		},
+		Commands: []*cli.Command{
+			{
+				Name:      "start",
+				Usage:     "Start a new Job.",
+				UsageText: "telepilot [global options] start <command> [arguments...]",
+				Action: func(ctx context.Context, cmd *cli.Command) error {
+					if cmd.Args().Len() == 0 {
+						return cli.ShowSubcommandHelp(cmd)
```

Comment:
The default behavior of the cli lib is to exit 0 after usage when missing arguments/subcommand. Returning here even without error maintains that consistency and allows for any potential cleanup to run instead of exiting right away.